### PR TITLE
Update 'implement interface' to forward explicit members to implicit ones

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/67023

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/ImplementInterface/ImplementInterfaceTests.cs [hunk 1]
```diff
@@ -11976,4 +12011,122 @@ static explicit I11<C11>.operator int(C11 x)
...
+                    {
+                        throw new NotImplementedException();
+                    }
+                }
+            
+                object IEnumerator.Current
+                {
+                    get
+                    {
+                        return Current;
```
Comment (ToddGrun) [inline src/Analyzers/CSharp/Tests/ImplementInterface/ImplementInterfaceTests.cs:12080]: >return Current; [](http://example.com/codeflow?start=24&length=15)

Is this desired? The boxing might not be necessary depending on how the Current method is implemented.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs [hunk 2]
```diff
@@ -209,12 +213,27 @@ public static ImmutableArray<SyntaxNode> GetGetAccessorStatements(
             return [generator.ReturnStatement(expression)];
         }
 
-        return preferAutoProperties ? default : generator.CreateThrowNotImplementedStatementBlock(compilation);
+        if (preferAutoProperties)
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs:216]: >if (preferAutoProperties) [](http://example.com/codeflow?start=8&length=25)

we don't want to do this for auto-properties when there are conflicting properties too?

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs [hunk 3]
```diff
@@ -235,9 +254,18 @@ public static ImmutableArray<SyntaxNode> GetSetAccessorStatements(
...
         }
 
-        return preferAutoProperties
-            ? default
-            : generator.CreateThrowNotImplementedStatementBlock(compilation);
+        if (preferAutoProperties)
+            return default;
+
+        // Forward from the explicit property we're creating to the existing property it conflicts with if possible.
+        if (conflictingProperty is { SetMethod: not null, Parameters.Length: 0 } &&
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs:261]: >Parameters.Length [](http://example.com/codeflow?start=58&length=17)

For my learning, setter properties don't have a parameter corresponding to "value"?

File: src/Analyzers/CSharp/Tests/ImplementInterface/ImplementInterfaceTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/ImplementInterface/ImplementInterfaceTests.cs:12080]: This is "fine".  it's the legacy apis calling to the modern ones.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs:216]: honestly.  i'm really trying to hit the sweet spot here.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs:261]: no.  that's likely a bug.

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs]: oh.  this was correct, but could be simplified.  the *property* has no parameters.  the *set accessor* has a single parameter.  i updated things accordingly.

**Comment by ToddGrun** [comment]
>                 return [generator.ExpressionStatement(generator.AssignmentStatement(generator.MemberAccessExpression(generator.ThisExpression(), property.Name), generator.IdentifierName("value")))];

Not from this PR, but should the "value" string be changed on line 252 too?

---
Refers to: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions.cs:265 in 0052955. [](commit_id = 0052955e6dacbee57171b3010dc89bfa0db40a32, deletion_comment = True)

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
> Not from this PR, but should the "value" string be changed on line 252 too?

Caring level is low.  I'm focusing on issues peoplee report.

**Pull request merged**